### PR TITLE
Setup Testing Structure

### LINKED ISSUE #27: Set up unit testing
We need a way to test components and code.
Setup the folder structure and add the utilities needed for testing.

### PR DESCRIPTION
**Pull request by Linkaloo**
closes #27 
This PR will create the testing structure for the server and the client.
This PR will also add all the dependencies and configuration needed.

On the Client side:
We will use Vitest, which is a testing framework that works with Vite.
We will use v8 for coverage.
The client side should be completely setup

On the Server side:
We will use Jest, a testing framework that works well with node.
It needs babel for ts usage so I have added that here as well.

For both:
There are 2 scripts added in each package file
- test: This runs all tests found
- coverage: This runs all tests found and creates a coverage report

coverage will create a coverage folder under the client or server. There is an index.html file there you can open. This folder is not synced into main.

Please follow this folder structure idea:

src and test at the same level.
then folder containing files (components or routes)
then {name of file}.test.ts
ensure that the {name of file} is the same as how it looks in src, except for casing.

Example:
For: src/controllers/UserController.ts
the test is: test/controllers/userController.test.ts

### CODE REVIEW

File: server/tsconfig.json
Comment (Linkaloo) [inline server/tsconfig.json:1]: I just cleaned up all the comments in this file

File: package.json
Comment (Linkaloo) [inline package.json:1]: This is the root level package file, this should not have any dependencies in it. However, if this causes issues for you guys for some reason let me know

**Review by KingJordan152 — APPROVED** [review]
Everything looks good. All configurations are appropriate and necessary to get testing working on both the frontend and backend.

**Review by tbmcallister — APPROVED** [review]
Framework looks good, and i think this will suit us well for adding in testing in the future.

Jest seems like a great way for us to perform testing, and develop our CI/CD

Looks good and good job!

**Pull request merged**